hydra-node/exe/hydra-node/Main.hs

```diff
@@ -31,23 +33,48 @@ import Hydra.Node (
   initEnvironment,
   runHydraNode,
  )
-import Hydra.Options (ChainConfig (..), LedgerConfig (..), Options (..), parseHydraOptions)
+import Hydra.Options (
+  ChainConfig (..),
+  Command (Publish, Run),
+  LedgerConfig (..),
+  Options (..),
+  PublishOptions (..),
+  parseHydraCommand,
+ )
 import Hydra.Party (Party)
 
 main :: IO ()
 main = do
-  o@Options{verbosity, host, port, peers, apiHost, apiPort, monitoringPort, chainConfig, ledgerConfig} <- identifyNode <$> parseHydraOptions
-  env@Environment{party} <- initEnvironment o
-  withTracer verbosity $ \tracer' ->
-    withMonitoring monitoringPort tracer' $ \tracer -> do
-      eq <- createEventQueue
-      withChain tracer party (putEvent eq . OnChainEvent) chainConfig $ \oc ->
-        withNetwork (contramap Network tracer) host port peers (putEvent eq . NetworkEvent) $ \hn ->
-          withAPIServer apiHost apiPort party (contramap APIServer tracer) (putEvent eq . ClientEvent) $ \server -> do
-            withCardanoLedger ledgerConfig $ \ledger -> do
-              node <- createHydraNode eq hn ledger oc server env
-              runHydraNode (contramap Node tracer) node
+  command <- parseHydraCommand
+  case command of
+    Run options ->
+      run (identifyNode options)
+    Publish options ->
+      publish options
```

Comment:
We now have two commands. Note that this is backward compatible (cf. changes Options module) and an explicit command is only needed for `publish-scripts`. The top-level `hydra-node` invocation will implicitly assume `run`